src/Tools/dotnet-dump/Analyzer.cs

```diff
@@ -0,0 +1,71 @@
+using Microsoft.Diagnostic.Utilities;
+using Microsoft.Diagnostics.Runtime;
+using System.Collections.Generic;
+using System.CommandLine;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostic.Tools.Dump
+{
+    public class Analyzer
+    {
+        private readonly KeyboardInput _keyboardInput;
+        private readonly CommandProcessor _commandProcessor;
+
+        public Analyzer()
+        {
+            _keyboardInput = new KeyboardInput();
+            _commandProcessor = new CommandProcessor(new Assembly[] { Assembly.GetEntryAssembly() });
+        }
+
+        public async Task<int> Analyze(FileInfo dumpFile, string[] commands)
+        {
+            _keyboardInput.Out.WriteLine($"Loading core dump: {dumpFile} ...");
```

Comment:
_keyboardInput.Out seems like an odd name. Perhaps 'KeyboardInput' should be renamed to something more generic such as 'shell' or 'terminal' to suggest that it has both input and output capabilities?

Reply:
The "KeyboardInput" class name is left over from where I stole that code.  You are right that it doesn't fit exactly. I'll come up with a better name.

Reply:
What about "ConsoleProvider" as a name?

Reply:
Fine by me : )